packages/react/src/components/MapCard/ScrollingControls.jsx

```diff
@@ -0,0 +1,125 @@
+import classnames from 'classnames';
+
+import { settings } from '../../constants/Settings';
+import { MapControlPropType } from '../../constants/CardPropTypes';
+
+const { iotPrefix } = settings;
+const BASE_CLASS_NAME = `${iotPrefix}--map-scroll-controls`;
+
+const propTypes = {
+  /** The list of controls placed in the scrollable group */
+  controls: PropTypes.arrayOf(MapControlPropType).isRequired,
+  scrollUpIconDescriptionText: PropTypes.string,
+  scrollDownIconDescriptionText: PropTypes.string,
+  tooltipPosition: PropTypes.string,
+  testId: PropTypes.string,
+  /** Number of visible items shown between the scroll buttons */
+  visibleItemsCount: PropTypes.number,
+};
+
+const defaultProps = {
+  visibleItemsCount: 6,
+  testId: 'map-scroll-controls',
+  tooltipPosition: 'left',
+  scrollUpIconDescriptionText: 'Scroll up',
+  scrollDownIconDescriptionText: 'Scroll down',
+};
+
+const ScrollingControls = ({
+  controls,
+  visibleItemsCount,
+  tooltipPosition,
+  scrollUpIconDescriptionText,
+  scrollDownIconDescriptionText,
+  testId,
+}) => {
+  const scrollContainerRef = useRef();
+  const buttonHeightPx = 40;
+  const scrollBy = (visibleItemsCount - 1 || 1) * buttonHeightPx;
+  const [scrollDownDisabled, setScrollDownDisabled] = useState(false);
+  const [scrollUpDisabled, setScrollUpDisabled] = useState(true);
+
+  const handleScrolling = (distance) => {
+    const container = scrollContainerRef.current;
+    const scrollingDown = distance > 0;
+
+    container.scrollBy({ left: 0, top: distance, behavior: 'smooth' });
+    setScrollUpDisabled(Math.abs(container.scrollTop) + distance === 0);
+    setScrollDownDisabled(
+      scrollingDown &&
+        container.scrollHeight - (Math.abs(container.scrollTop) + scrollBy) ===
+          container.clientHeight
+    );
+  };
+
+  return (
+    <div data-testid={testId} className={classnames(`${BASE_CLASS_NAME}__container`)}>
+      <Button
+        data-testid={`${testId}-scroll-up`}
+        className={classnames(`${BASE_CLASS_NAME}__btn`, `${BASE_CLASS_NAME}__scroll-btn`)}
+        disabled={scrollUpDisabled}
+        kind="ghost"
+        size="field"
+        hasIconOnly
+        tooltipPosition={tooltipPosition}
+        renderIcon={ChevronUp32}
+        iconDescription={scrollUpIconDescriptionText}
+        onClick={() => handleScrolling(scrollBy * -1)}
+      />
+      {scrollUpDisabled ? null : <div className={`${BASE_CLASS_NAME}__gradient`} />}
+      <div
+        data-testid={`${testId}-scroll-area`}
+        ref={scrollContainerRef}
+        style={{
+          [`--visble-items-count`]: visibleItemsCount,
+          [`--scroll-area-height`]: `${visibleItemsCount * buttonHeightPx}px`,
+        }}
+        className={classnames(`${BASE_CLASS_NAME}__scroll-area`)}
+      >
+        {controls.map((control, i) => {
+          return (
+            <Button
+              className={`${BASE_CLASS_NAME}__btn`}
+              key={`${control.iconDescription}-${i}`}
```

Comment:
Should consider using something unique to the value rather than the index

Reply:
See previous comment